tests/unit/preprocessor/_volume/test_volume.py

```diff
@@ -44,7 +44,8 @@ def setUp(self):
             [0.5, 5., 50.],
             standard_name='depth',
```

Comment:
you may want to add a separate test for `air_pressure` as well, or even better, change depth with a generic Z-axis coordinate `zcoord` and guess it (so that the attr axis gets added to the coordinate) like this:
```
self.cube.coord('zcoord').attributes['positive'] = 'up'
iris.util.guess_coord_axis(self.cube.coord('zcoord'))
```

Reply:
Hmmm.....
```
           ValueError: 'zcoord' is not a valid standard_name
```

Reply:
stay put, am fixing it straight in the PR now :radio:

Reply:
ok done, the thing is - `extract_volume` now works with __any__ generic z-coord without naming it explicitly (which is very good) but all the other functions need an explicit name of a zcoordinate - do you need to keep that behaviour or we can nuke all the coordz arguments and replace them with axis='z' in the function body?

Reply:
ah sod it, too much work :grin:

Reply:
actually I've done it :grin: now all those functions will work with any z coord without explicitly naming it - did I just kill all your oceans recipes?

Reply:
yes...

Reply:
feel free to revert, but I think we're in a better position now, both from a generality point of view and from user perspective (ie they don't have to dig in their data to figure out what the name of the vertical axis is)

Reply:
I can fix all the func args in the diags if you want me to :smiling_imp:

Reply:
Also all the documentation will need to be re-written.

Reply:
I did change the docstrings though :grin:

Reply:
Here's a question for you. If we can automatically figure out the name of the z direction, could we also do that with latitude and longitude? 

Do we need to  have coord1 and coord2 in (for instance):
```
    average_region:
      coord1: longitude
      coord2: latitude
```
?

Reply:
yes, in fact, we can get rid of all the explicit calls to latitude and longitude axes as well, in the same manner, wherever we collapse on all the values; we should use them only when we apply specific boundaries or individual operations on individual axes

Reply:
but we need to think carefully if we want to support anything that's not CMOR standard 9which we don't) since these axes = X, Y, Z and T are present only in CMORized files

Reply:
That must surely be for a future issue/PR though. Changing coordz was already well beyond the scope of this PR!

Reply:
yup, absolutely!